src/java.base/share/classes/java/net/Socket.java

```diff
@@ -1020,8 +1020,8 @@ public void setSoLinger(boolean on, int linger) throws SocketException {
             if (linger < 0) {
                 throw new IllegalArgumentException("invalid value for SO_LINGER");
             }
-            if (linger > 65535)
-                linger = 65535;
+            if (linger > 32767)
```

Comment:
I agree reducing the linger value to 32767 is valid for this test. The test is not verifying the maximum linger value but rather just verifying if setting the linger value works.

The docs imply the maximum value is 65535:

https://docs.oracle.com/en/java/javase/12/docs/api/java.base/java/net/SocketOptions.html#SO_LINGER

but I'm not sure if that is a requirement for passing certification.

Reply:
If it is, then we'll be unable to do so without some sort of egregious hack since that's impossible on either FreeBSD or OpenBSD.  At the moment this is also a bug for our users since values larger than 32767 are going to wrap around to a negative value, so I'm inclined to merge this one.